mne/preprocessing/ica.py

```diff
@@ -1048,6 +1048,15 @@ def find_bads_ecg(self, inst, ch_name=None, threshold=None, start=None,
             if threshold is None:
                 threshold = 0.25
             if isinstance(inst, BaseRaw):
+                reject_by_annotation = 'omit' if reject_by_annotation else None
+                start, stop = _check_start_stop(inst, start, stop)
+
+                # update the instance with new data
+                _data = inst.get_data(picks=None, start=start, stop=stop,
+                                      reject_by_annotation=reject_by_annotation)
+                inst._data = _data  # Could be done directly?
```

Comment:
This is a bit dangerous because things like `raw.times` will not be updated properly. A better fix (and one that avoids the preloading that this forces) is probably to make `create_ecg_epochs` capable of handling Annotations properly.

Reply:
You are right. We need to make changes in create_ecg_epochs to get a more proper solution.
I will come up with a revised version soon.